frmts/vrt/vrtprocesseddataset.cpp

```diff
@@ -521,6 +538,66 @@ CPLErr VRTProcessedDataset::Init(const CPLXMLNode *psTree,
         return CE_Failure;
     }
 
+    // Use only up to 40% of RAM to acquire source bands and generate the output
+    // buffer.
+    const auto nMaxMemAllowed = CPLGetUsablePhysicalRAM() / 10 * 4;
```

Comment:
This behavior seems similar to the `-wm` argument of `gdalwarp`. Would it makes sense to have a configuration option like `GDAL_WORKING_MEMORY` that can be used in multiple places? Or would that be too confusing if, for example, our source for the processed VRT is a warped VRT?

Reply:
> Or would that be too confusing if, for example, our source for the processed VRT is a warped VRT?

yes there's indeed a risk of confusion. One could imagine there could be some memory allocator in GDAL that would know which components of GDAL need memory allocation and distribute what is available amongst them. Like component A would claim SIZE_A , B would claim SIZE_B, etc.  and then if sume(SIZE_i)> max_allowed, the allocator would tell X it just can use SIZE_X * max_allowed / sum(SIZE_i). But I don't see how that could be implemented in our architecture.